data/data/aws/cluster/vpc/vpc-public.tf

```diff
@@ -94,4 +94,7 @@ resource "aws_nat_gateway" "nat_gw" {
     },
     var.tags,
   )
+
+  # https://issues.redhat.com/browse/OCPBUGS-891
+  depends_on = [aws_eip.nat_eip]
```

Comment:
What if we add `aws_subnet.public_subnet` too to this to prevent future errors?

Reply:
I agree with Aditya, we _could_ add that, but I am worried it may not work. Azure has/had something very similar which I initially added a depends_on, but ultimately it failed. I linked a PR below to what was completed for Azure. Not promising this works or is what we need but it is something to consider.

https://github.com/openshift/installer/pull/6385

Reply:
That PR didn't fix it (basing it off of this comment https://coreos.slack.com/archives/C042E82T490/p1663855691063339) so might need to think of something else